src/overlays/actors/ovl_En_GeldB/z_en_geldb.c

```diff
@@ -33,93 +97,1466 @@ const ActorInit En_GeldB_InitVars = {
+        this->actor.gravity = 0.0f;
+        func_80033260(globalCtx, &this->actor, &this->rightFootPos, 3.0f, 2, 2.0f, 0, 0, 0);
+        func_80033260(globalCtx, &this->actor, &this->leftFootPos, 3.0f, 2, 2.0f, 0, 0, 0);
+    }
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.y, this->actor.groundY + 300.0f, 1.0f, 20.5f, 0.0f);
+        if (--this->timer == 0) {
+            Actor_Kill(&this->actor);
+        }
+    }
+}
+
+void EnGeldB_SetupReady(EnGeldB* this) {
+    SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_0600B6D4, -4.0f);
+    this->actionState = GELDB_READY;
+    this->timer = Math_Rand_ZeroOne() * 10.0f + 5.0f;
+    this->actor.speedXZ = 0.0f;
+    this->actor.posRot.rot.y = this->actor.shape.rot.y;
+    EnGeldB_SetupAction(this, EnGeldB_Ready);
+}
+
+void EnGeldB_Ready(EnGeldB* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
+    s16 angleToLink;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->lookTimer != 0) {
+        angleToLink = this->actor.yawTowardsLink - this->actor.shape.rot.y - this->headRot.y;
+        if (ABS(angleToLink) > 0x2000) {
+            this->lookTimer--;
+            return;
+        }
+        this->lookTimer = 0;
+    }
+    angleToLink = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+    if (!EnGeldB_DodgeRanged(globalCtx, this)) {
+        if (this->unkTimer != 0) {
+            this->unkTimer--;
+
+            if (ABS(angleToLink) >= 0x1FFE) {
+                return;
+            }
+            this->unkTimer = 0;
+        } else if (EnGeldB_ReactToPlayer(globalCtx, this, 0)) {
+            return;
+        }
+        angleToLink = player->actor.shape.rot.y - this->actor.shape.rot.y;
+        if ((this->actor.xzDistFromLink < 100.0f) && (player->swordState != 0) && (ABS(angleToLink) >= 0x1F40)) {
+            this->actor.shape.rot.y = this->actor.posRot.rot.y = this->actor.yawTowardsLink;
+            EnGeldB_SetupCircle(this);
+        } else if (--this->timer == 0) {
+            if (func_8002E084(&this->actor, 30 * 0x10000 / 360)) {
+                if ((210.0f > this->actor.xzDistFromLink) && (this->actor.xzDistFromLink > 150.0f) &&
+                    (Math_Rand_ZeroOne() < 0.3f)) {
+                    if (func_80033AB8(globalCtx, &this->actor) || (Math_Rand_ZeroOne() > 0.5f) ||
+                        (ABS(angleToLink) < 0x38E0)) {
+                        EnGeldB_SetupRollForward(this);
+                    } else {
+                        EnGeldB_SetupSpinAttack(this);
+                    }
+                } else if (Math_Rand_ZeroOne() > 0.3f) {
+                    EnGeldB_SetupAdvance(this, globalCtx);
+                } else {
+                    EnGeldB_SetupCircle(this);
+                }
+            } else {
+                EnGeldB_SetupPivot(this);
+            }
+            if ((globalCtx->gameplayFrames & 0x5F) == 0) {
```

Comment:
% 96 possibly?

Reply:
Functionally different. `& 0x5F` hits on frames 0 and 32 of a 128 frame cycle, while `% 96` hits on frame 0 of a 96 frame cycle.